### ISSUE TITLE
Area chart SVG contains duplicate `fill-opacity` attribute

### ISSUE THREAD

**Issue by pratik136** [highcharts/highcharts#5366]
#### Expected behaviour

Using the API method `chart.getSVGForExport()` the resultant SVG should not contain duplicate attributes on the same element. 
#### Actual behaviour

The generated SVG contains the `<path>` element with duplicated `fill-opacity` attribute. 
#### Live demo with steps to reproduce

http://jsfiddle.net/msteichen/o4a2e7Lp/
#### Affected browser(s)

All Browsers.

**Comment by jon-a-nygaard** [comment]
Thanks for reporting, we will fix this issue as soon as possible.

 _Internal Note: the issue occurs in the function `sanitizeSVG` located in the exporting module. This function replaces rgba attributes with rgb and fill-opacity attributes. Can probably be fixed by adding a sort of negative lookahead to the regex_

**Comment by TorsteinHonsi** [comment]
@jon-a-nygaard A cheaper solution, just don't add the `fill-opacity` attribute when it's 1?

**Comment by jon-a-nygaard** [comment]
## Workaround
For those in need of a workaround, adding the snippet below after loading Highcharts will fix the issue.
```javascript
Highcharts.wrap(Highcharts.Chart.prototype, 'sanitizeSVG', function(proceed, svg) {
  var each = Highcharts.each;
  svg = svg.replace(/\<[^<]+(fill|stroke)="rgba[^<]+>/g, function(match) {
    var tag = match;
    each(['stroke', 'fill'], function(a) {
      var opacity,
        exp;
      // Convert rgba to rgb
      if (match.indexOf(a + '="rgba') > -1) {
        exp = new RegExp(a + '="rgba\\(([ 0-9]+,[ 0-9]+,[ 0-9]+),([ 0-9\.]+)\\)"');
        tag = tag.replace(exp, function(str, p1, p2) {
          opacity = parseFloat(p2);
          return a + '="rgb(' + p1 + ')"';
        });
        if (opacity !== 1) {
          exp = new RegExp(' ' + a + '-opacity="([^"]+)"');
          // Get new opacity value, and remove attribute from string.
          tag = tag.replace(exp, function(str, p1) {
            opacity *= parseFloat(p1);
            return '';
          });
          // Append new opacity
          tag = tag.slice(0, -1) + ' ' + a + '-opacity="' + opacity + '">';
        }
      }
    });
    return tag;
  });
  svg = proceed.apply(this, [svg]);
  return svg;
});
```

**Comment by jon-a-nygaard** [comment]
The issue is connected to the fix for #3095. Once rgba is 100% supported everywhere, the code below can be removed, and the issue will resolve it self.
```javascript
// Batik doesn't support rgba fills and strokes (#3095)
.replace(/(fill|stroke)="rgba\(([ 0-9]+,[ 0-9]+,[ 0-9]+),([ 0-9\.]+)\)"/g, '$1="rgb($2)" $1-opacity="$3"')
```

**Comment by smithk58** [comment]
For someone using the temporary fix posted a couple of comments up, you may need to modify it slightly. Some browsers (e.g. ie11) seem to use self closing path tags  (i.e. <path />), and the above code assumes they don't (i.e. assumes <path> </path>). This can result in path tags that are replaced ending like down below (notice how the fill-opacity was added after the /):
```
 / fill-opacity="0.8">
```

The problem part is just below the 'Append new opacity' comment in the code, which trims off the last character and appends the opacity and a new carrot. Below is the change I made to check if it's a self closing tag and adjust appropriately.
```
var isSelfClosing = tag.substr(tag.length -2) === '/>';
    	        	  tag = tag.slice(0, isSelfClosing ? -2 : -1) + ' ' + a + '-opacity="' + opacity + (isSelfClosing ? '"/>' : '">');
```

Some svg interpreters seem to handle it fine even without this fix (not sure if they just toss out the last attribute or what), but others throw a fit about your path tag not being closed.
